README.md

```diff
@@ -139,6 +139,60 @@ To run the tests, execute the following command from the repo's root directory:
 ./scripts/run.e2e.sh
 ```
 
+## Snapshots usage for local networks
+
+Network snapshots are used by the CLI in order to keep track of blockchain state, and to improve performance of local operations.
+
+They are the main way to persist subnets, blockchains, and blockchain operations, among different executions of the tool.
+
+Three different kind of snapshots are used:
+- The `bootstrap snapshot` is provided as the starting network state. It is never modified by CLI usage. 
+Designed for fast deploys. Enables full reset of the blockchain state.
+- The `default snapshot` is the main way to keep track of blockchain state. Used by default in the tools.
+It is initialized from the `bootstrap snapshot`, and after that is updated from CLI operations.
+- `custom snapshots` can be specified by the user, to save and restore particular states. Only changed if
+explicitely asked to do so.
+
+### Local networks
+
+Usage of network for local operations:
+- The local network will be started only if it not already running
+- If the network is not running, both `network start` and `subnet deploy` will start it from the `default snapshot`. `subnet deploy`
+will also do the deploy on the started network.
+- If the network is running, `network start` will do nothing, and `subnet deploy` will use the running one to do the deploy.
+- The local network will run until calling `network stop`, `network clean`, or until machine reboot
+
+### Default snapshot
+
+Changes in `default snapshot` by local operations:
+- First call of `network start` or `subnet deploy` will initialize `default snapshot` from the `bootstrap snapshot`
+- Subnecuent calls to `subnet deploy` does not change the snapshot, only the running network
+- `network stop` persist the running network into the `default snapshot`
+- `network clean` copy again the `bootstrap snapshot` into the `default snapshot`, doing a reset of the state
+
+So tipically a user will want to do the deploy it needs, change the blockchain state in a specific way, and 
+after that execute `network stop` to preserve all the state. In a different session, `network start` or `subnet deploy`
+will recover that state.
+
+### Custom snapshots
+
+Changes in `custom snapshots` by local operations:
+- `network stop` can be given an optional snapshot name, will be used instead of the default one, to save the state
```

Comment:
I suggest for all these:
...can be given an optional snapshot name. This will then be used instead of the default one to save the state (no comma)